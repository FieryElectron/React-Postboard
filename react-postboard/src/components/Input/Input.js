import './Input.css';

const Input = ({ placeholder, type , onChange}) => {
    return (
        <div data-testid='inputContainer' className='inputContainer'>
            <input data-testid='input' className="input" placeholder={placeholder} type={type} onChange={onChange} spellCheck="false"/>
        </div>
    )
}

Input.defaultProps = {
    text: '',
    placeholder: 'placeholder',
    type: 'text'
}

export default Input;