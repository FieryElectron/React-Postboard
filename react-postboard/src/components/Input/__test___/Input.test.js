import ReactDOM from "react-dom/client";
import { render, fireEvent  } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { act } from 'react-dom/test-utils';
import '@testing-library/jest-dom/extend-expect';

import Input from '../Input';

let container;

beforeEach(() => {
  container = document.createElement('div');
  document.body.appendChild(container);
});

afterEach(() => {
  document.body.removeChild(container);
  container = null;
});

it("renders input without crashing", ()=>{
    act(() => {
        ReactDOM.createRoot(container).render(<Input />);
    });
});

it("renders input class correctly", ()=>{
    const {getByTestId} = render(<Input/>);
    expect(getByTestId('inputContainer')).toHaveAttribute('class', 'inputContainer');
    expect(getByTestId('input')).toHaveAttribute('class', 'input');
});

it("renders input placeholder correctly", ()=>{
    const {getByTestId} = render(<Input placeholder="Input some text"/>);
    expect(getByTestId('input')).toHaveAttribute('placeholder', 'Input some text');
});

it("renders input type correctly", ()=>{
    const {getByTestId} = render(<Input type="Password"/>);
    expect(getByTestId('input')).toHaveAttribute('type', 'Password');
});

it("should call onChange callback", ()=>{
    const onChange = jest.fn();
    const {getByTestId} = render(<Input onChange={onChange}/>);
    fireEvent.change(getByTestId('input'), {target: {value: 'some text'}});
    expect(onChange).toHaveBeenCalled();
});

it("matchs snapshot 1", ()=>{
    const onChange = jest.fn();
    const tree = renderer.create(<Input placeholder='placeholder' type="Password" onChange={onChange}/>).toJSON();
    expect(tree).toMatchSnapshot();
}); 

it("matchs snapshot 2", ()=>{
    const onChange = jest.fn();
    const tree = renderer.create(<Input placeholder='username' onChange={onChange}/>).toJSON();
    expect(tree).toMatchSnapshot();
}); 

