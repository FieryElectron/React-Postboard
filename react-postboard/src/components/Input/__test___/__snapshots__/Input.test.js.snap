// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matchs snapshot 1 1`] = `
<div
  className="inputContainer"
  data-testid="inputContainer"
>
  <input
    className="input"
    data-testid="input"
    onChange={[MockFunction]}
    placeholder="placeholder"
    type="Password"
  />
</div>
`;

exports[`matchs snapshot 2 1`] = `
<div
  className="inputContainer"
  data-testid="inputContainer"
>
  <input
    className="input"
    data-testid="input"
    onChange={[MockFunction]}
    placeholder="username"
    type="text"
  />
</div>
`;
